scripts/nb-tester/test-notebook.py

```diff
@@ -35,6 +36,34 @@
 ]
 
 
+def filter_paths(paths: list[Path], submit_jobs: bool) -> Iterator[Path]:
+    """
+    Filter out any paths we don't want to run, printing messages.
+    """
+    for path in paths:
+        if path.suffix != ".ipynb":
+            print(f"ℹ️ Skipping {path}; file is not `.ipynb` format.")
+            continue
+
+        if any(path.match(glob) for glob in NOTEBOOKS_EXCLUDE):
+            this_file = Path(__file__).relative_to(Path(".").resolve())
+            print(
+                f"ℹ️ Skipping {path}; to run it, edit `NOTEBOOKS_EXCLUDE` in {this_file}."
+            )
+            continue
+
+        if (
+            any(path.match(glob) for glob in NOTEBOOKS_THAT_SUBMIT_JOBS)
+            and not submit_jobs
```

Comment:
Nit: don't waste time doing an O(n) operation if the boolean is false anyways. n is so small this is nitty, though.

```suggestion
            not submit_jobs
            and any(path.match(glob) for glob in NOTEBOOKS_THAT_SUBMIT_JOBS)
```